src/Features/CSharp/Portable/Formatting/CSharpUseProgramMainNewDocumentFormattingProvider.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.ConvertProgram;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Microsoft.CodeAnalysis.CSharp.Formatting
+{
+    [ExportNewDocumentFormattingProvider(LanguageNames.CSharp), Shared]
+    internal class CSharpUseProgramMainNewDocumentFormattingProvider : INewDocumentFormattingProvider
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public CSharpUseProgramMainNewDocumentFormattingProvider()
+        {
+        }
+
+        public async Task<Document> FormatNewDocumentAsync(Document document, Document? hintDocument, CancellationToken cancellationToken)
+        {
+            var options = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
+            var option = options.GetOption(CSharpCodeStyleOptions.PreferTopLevelStatements);
+            if (option.Value)
+                return document;
+
+            return await ConvertProgramHelpers.ConvertToProgramMainAsync(document, cancellationToken).ConfigureAwait(false);
+        }
```

Comment:
Should we do that *only if* the new document is top-level (ie, only happens when creating a new project where `.editorconfig` already existed preferring one way to the other?)

Reply:
the default is to prefer top level statements :)